src/bls/bls.h

```diff
@@ -507,6 +508,14 @@ class CBLSLazyWrapper
 
     bool operator==(const CBLSLazyWrapper& r) const
     {
+        // If neither bufValid or objInitialized are set, then the object is the default object.
+        const bool is_default{bufValid || objInitialized};
+        const bool r_is_default{r.bufValid || r.objInitialized};
```

Comment:
This feels broken
```suggestion
        const bool is_default{!bufValid && !objInitialized};
        const bool r_is_default{!r.bufValid && !r.objInitialized};
```

Reply:
Yes, you are correct! @knst's patch messed me up! :D I should've seen it tho 🤷